### PR TITLE
[GENFI] Handle T1 converted as T2

### PR DESCRIPTION
**Pull request by MatthieuJoulot**
Some T1 are not correctly converted. This is due to the fact that the conversion is based on the series description and that for a large amount of T1, it is wrongly described as a T2 (not the other way around, as far as we know). `dcm2niix` makes the same mistake as well.
Therefore, we if the `SeriesDescription` is T1 or T2, we check the `EchoTime` and `RepetitionTime` that characterise T1 or T2.

This PR will need a change in the data CI because of that.

### CODE REVIEW

**Comment by MatthieuJoulot** [comment]
The values 60 and 2100 have been arbitrarily chosen based on the following document and on the values found in the sequences.
Image Pasted at 2023-5-30 15-20

**Comment by MatthieuJoulot** [comment]
The data CI PR is available [here](https://github.com/aramis-lab/clinica_data_ci/pull/45)

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:65]: Should this be removed ?

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:807]: These changes should not be there I believe. They belong to a previously merged PR.

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:86]: By curiosity, what is the type of `series_description` ?

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:92]: You explained these magic numbers in the PR but not in the code such that they will still seem pretty magical in a few months/years for someone reading this...

I would both use constant with explicit names (`MAX_ECHO_TIME_FOR_A_T1` and `MAX_REPETITION_TIME_FOR_A_T1` maybe???) and put the explanations you wrote in the PR in a docstring of this function.

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:87]: Does it happen in practice to have mixtures of upper and lower case ?

This feels a little weird when reading it. I would expect the value(s) in `series_description` to be made lower case and then compared to "t1" and/or "t2". Does this make sense ?

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @MatthieuJoulot !
Looks mostly good besides the problematic rebase.
I just have a few suggestions and questions.

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (MatthieuJoulot) [reply clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:86]: `SeriesDescription` is a string

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:96]: Is it an issue that the function returns only "t1" in this case, but the full series description in other cases ?

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:82]:
```suggestion
    are supposed to be very different between a T1 and T2. Therefore, we use this proxy to check that the series description is coherent.
```

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (MatthieuJoulot) [reply clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:96]: No it is not. The function `identify_modality` looks for t1 in the series description

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @MatthieuJoulot !
Let's wait for the CI to finish with the updated data and I'll merge if green.

**Pull request merged**
